src/main/java/mate/academy/dao/impl/ManufacturerDaoJdbcImpl.java

```diff
@@ -0,0 +1,120 @@
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(parseDataFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong."
+                    + " Can't get ID dy index " + manufacturerId, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String query = "UPDATE taxi_service "
+                + "SET name = ?, country = ? "
+                + "WHERE id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.setLong(3, manufacturer.getId());
+            statement.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong."
+                    + " Can't update manufacturer " + manufacturer, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long manufacturerId) {
+        String query = "UPDATE taxi_service "
+                + "SET deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            //statement.setBoolean(1, true);
+            statement.setLong(1, manufacturerId);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. "
+                    + "Can't delete manufacturer " + manufacturerId, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Manufacturer manufacturer) {
+        return deleteById(manufacturer.getId());
+    }
+
+    @Override
+    public List<Manufacturer> getAllManufactures() {
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        String query = "SELECT * FROM taxi_service "
+                + "WHERE deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                manufacturers.add(parseDataFromResultSet(resultSet));
+            }
+            return manufacturers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. "
+                    + "Can't get all manufacturers", e);
+        }
+    }
+
+    private Manufacturer parseDataFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getLong("id");
```

Comment:
see checklist - we should use `getObject()` here

Reply:
Ok. Sorry, next time I'll try to be more attentive.